src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/CalendarEvents/Queries/GetCalendarEventById/GetCalendarEventByIdQueryHandler.cs

```diff
@@ -0,0 +1,20 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeAan.Application.Infrastructure;
+
+namespace SFA.DAS.ApprenticeAan.Application.CalendarEvents.Queries.GetCalendarEventById;
+
+public class GetCalendarEventByIdQueryHandler : IRequestHandler<GetCalendarEventByIdQuery, GetCalendarEventByIdQueryResponse>
+{
+    private readonly IAanHubRestApiClient _apiClient;
+
+    public GetCalendarEventByIdQueryHandler(IAanHubRestApiClient apiClient)
+    {
+        _apiClient = apiClient;
+    }
+
+    public async Task<GetCalendarEventByIdQueryResponse> Handle(GetCalendarEventByIdQuery command, CancellationToken cancellationToken)
+    {
+        var result = await _apiClient.GetCalendarEventById(command.CalendarEventId, command.RequestedByMemberId, cancellationToken);
```

Comment:
The check for status code should be done here. There are only two possibilities, record found or not found. If record is found we return the content  (CalendarEventDetails) otherwise you return a null. In the controller null should be interpreted as NotFound.

Reply:
Shouldn't we return BadRequest if the X-RequestedByMemberId header is empty or invalid?

Reply:
Think front end, will FE make a request with invalid header?